### DISCUSSION TITLE
nginx proxy manager bad gateway 502 New Install

### DISCUSSION THREAD

**Discussion by ResearchForumOnline** [NginxProxyManager/nginx-proxy-manager#2817] [Q&A] [answered]
Installed Ubuntu 20,18,22 all failed when i went to the login web ui page: bad gateway in red shows and can not login using default admin@example.com changeme
Is this a major issue as i have seen a few sites mention the issue for new installs with no actual easy understandable solution yet?
Thanks for replying if anyone does.

**Comment by truthsword** [comment]
v2.10 broke stuff ("bad gateway" is a symptom) Try 2.9.22.

**Reply by ResearchForumOnline** [reply]
Thanks fixed using a better docker-compose file.

**Comment by ResearchForumOnline** [comment, marked as answer]
Fixed using this:


 nano docker-compose.yml



File contents:



version: '3.8'
services:
  app:
    image: 'jc21/nginx-proxy-manager:latest'
    restart: unless-stopped
    ports:
      # These ports are in format <host-port>:<container-port>
      - '80:80' # Public HTTP Port
      - '443:443' # Public HTTPS Port
      - '81:81' # Admin Web Port
      # Add any other Stream port you want to expose
      # - '21:21' # FTP
    environment:
      # Mysql/Maria connection parameters:
      DB_MYSQL_HOST: "db"
      DB_MYSQL_PORT: 3306
      DB_MYSQL_USER: "npm"
      DB_MYSQL_PASSWORD: "SecurePasswordHere'"
      DB_MYSQL_NAME: "npm"
      # Uncomment this if IPv6 is not enabled on your host
      # DISABLE_IPV6: 'true'
    volumes:
      - ./data:/data
      - ./letsencrypt:/etc/letsencrypt
    depends_on:
      - db

  db:
    image: 'jc21/mariadb-aria:latest'
    restart: unless-stopped
    environment:
      MYSQL_ROOT_PASSWORD: 'SecurePasswordHere'
      MYSQL_DATABASE: 'npm'
      MYSQL_USER: 'npm'
      MYSQL_PASSWORD: 'SecurePasswordHere'
    volumes:
      - ./mysql:/var/lib/mysql





Then do:   docker-compose up -d



upgrading:
docker-compose pull
docker-compose up -d
.......................

**Reply by truthsword** [reply]
The only change I see to my compose, is this single line...

`./mysql:/var/lib/mysql`

I have been using

`./data/mysql:/var/lib/mysql`

Is that the change that fixes things? If so, 

Before I run `docker-compose up -d`, must I move my current `./data/mysql` contents to `./mysql` or can I start with an empty `./mysql` directory, and not lose my data, redirects, certs, etc.

Thank you!

**Reply by ResearchForumOnline** [reply]
hmm i am not sure, it could break it, if it is a new install then try it, i installed using ubuntu 20, docker

**Reply by truthsword** [reply]
This is an existing installation. I had a non-recoverable "Bad Gateway" experience a year ago on an RPi, and I don't want to relive that.

**Reply by ResearchForumOnline** [reply]
I wrote this for myself and others just now:
https://github.com/NginxProxyManager/nginx-proxy-manager/discussions/2827
